crates/services/executor/src/executor.rs

```diff
@@ -689,41 +615,32 @@ where
                 coinbase_contract_id,
                 gas_price,
                 execution_data,
-                execution_kind,
                 &mut tx_st_transaction,
             )?;
             tx_st_transaction.commit()?;
             tx
         };
 
-        block.transactions.push(tx);
+        block.transactions.push(tx.clone());
```

Comment:
It may be too expensive to clone transaction. You could return a reference to it. Like `block.transactions.as_ref().expect("We just added transaction")` instead.

Reply:
Turns out we don't need it because we are inspecting the txs after.